### PR TITLE
Assemble tinybones

### PR DESCRIPTION
**Pull request by 5han7anu-S**
Created a main function and integrated GraphQL with async processes so multiple instances can run simultaneously.

### CODE REVIEW

**Review by GraysonNocera — CHANGES_REQUESTED** [review]
I'm so sorry to be that guy, but can you see why the tests are failing? By tests, I mean the CI pipeline (pipeline.yaml) file.

File: experiments/basic_skeleton.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
+import { url_class } from '../src/url_class';
+
+// Given a Repo URL, uses REGEX to grab the repo and owner name
+// I **THINK** it works with any (GitHub) URL they can throw at us, but I haven't exhaustively tested it
+// To test the REGEX function, compile and run regextext.ts with different inputs.
+
+function getGitRepoDetails(url: string): {username: string, repoName: string} | null {
```
Comment (GraysonNocera) [inline experiments/basic_skeleton.ts:7]: I believe we could use a library for this instead. to reduce the amount of code we're writing. 

I think Koltan mentioned https://github.com/jonschlinkert/parse-github-url as a possible library for this. Unless you are very confident that this regex always works.

File: src/main.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
+"use strict";
```
Comment (GraysonNocera) [inline src/main.js]: Let's try to not have compiled .js files in scr/ and just keep our ts in there. If you type tsc, it should compile everything and put the compiled .js files in dist/ folder

File: src/parse_funcs.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
+import axios, { AxiosResponse } from "axios";
+import isGitHubUrl = require("is-github-url");
```
Comment (GraysonNocera) [inline src/parse_funcs.ts]: I was getting an import error here. Where is that function located?

**Review by GraysonNocera — CHANGES_REQUESTED** [review]

File: experiments/basic_skeleton.ts (hunk 1 above)
Comment (5han7anu-S) [reply experiments/basic_skeleton.ts:7]: I installed it and it had 4 security vulnerabilities, so I decided to do it manually. Works for everything I've tested so far

File: src/main.js (hunk 2 above)
Comment (5han7anu-S) [reply src/main.js]: Yeah, that's my bad. Usually, I manually delete the .js files.

tsc src/*.ts does not actually put the .js files in the ./dist folder. I don't know why. I've tried different edits on the TSConfig, but it doesn't work.

File: src/parse_funcs.ts (hunk 3 above)
Comment (5han7anu-S) [reply src/parse_funcs.ts]: It's a dependency. Node Module should be installed from npm.

Either of 

npm install
or
npm install is-github-url axios

should work

File: experiments/basic_skeleton.ts (hunk 1 above)
Comment (GraysonNocera) [reply experiments/basic_skeleton.ts:7]: Ok good call

File: src/main.js (hunk 2 above)
Comment (GraysonNocera) [reply src/main.js]: Yeah I just ran tsc instead and it was working. It also compiles everything in experiments/ too but as long as none of those files are broken it's fine. But you have to modify the rootDirs field in tsconfig.json for that to work I believe. But the correct configuration is in the main branch I believe so no need to worry about it much.

File: src/parse_funcs.ts (hunk 3 above)
Comment (GraysonNocera) [reply src/parse_funcs.ts]: Ok

**Comment by GraysonNocera** [comment]
Alright, so if you could delete those .js files in tsc, and fix the errors whenever the pipeline is run, I can approve and merge. The issue could be that I changed the name of the url_class to be Runner (sorry).

**Pull request closed without merging**